_studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw_ddi.h

```diff
@@ -185,22 +185,27 @@ typedef struct tagENCODE_CAPS_VP9
         mfxU16 BitSizeForSegmentation;
     };
 
-    inline void AddSeqHeader(mfxU32 width,
+    inline mfxStatus AddSeqHeader(mfxU32 width,
         mfxU32 height,
         mfxU32 FrameRateN,
         mfxU32 FrameRateD,
         mfxU32 numFramesInFile,
         mfxU8* pBitstream,
         mfxU32 bufferSize)
     {
+    	if (bufferSize == 0)
+    		return MFX_ERR_MORE_DATA;
+
         mfxU32   ivf_file_header[8] = { 0x46494B44, 0x00200000, 0x30395056, width + (height << 16), FrameRateN, FrameRateD, numFramesInFile, 0x00000000 };
-        memcpy_s(pBitstream, bufferSize, ivf_file_header, sizeof (ivf_file_header));
+        std::copy(std::begin(ivf_file_header),std::end(ivf_file_header), pBitstream);
+
+        return MFX_ERR_NONE;
     };
 
     inline void AddPictureHeader(mfxU8* pBitstream, mfxU32 bufferSize)
     {
         mfxU32 ivf_frame_header[3] = { 0x00000000, 0x00000000, 0x00000000 };
-        memcpy_s(pBitstream, bufferSize, ivf_frame_header, sizeof (ivf_frame_header));
+        std::copy(std::begin(ivf_frame_header),std::end(ivf_frame_header), pBitstream);
```

Comment:
std::fill_n(begin, 12, 0) better?

Reply:
yeah, but the task is h265e- use C++ primitives (memcpy => std::copy)
Is it fine to change memcpy => std::fill_n?